File: demo/src/main/java/com/nshmura/strictmodenotifier/demo/ClassInstanceLimitActivity.java
package com.nshmura.strictmodenotifier.demo;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;

public class ClassInstanceLimitActivity extends AppCompatActivity {

  private static final String EXTRA_COUNT = "EXTRA_COUNT";

  public static void start(Context context, int count) {
    Intent starter = new Intent(context, ClassInstanceLimitActivity.class);
    starter.putExtra(EXTRA_COUNT, count);
    context.startActivity(starter);
  }

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    // Post a message and delay its execution for 10 minutes.
    new Handler().postDelayed(new Runnable() {
      @Override public void run() {
      }
    }, 1000 * 60 * 10);

    int count = getIntent().getIntExtra(EXTRA_COUNT, 0);
    if (count > 0) {
      start(this, --count);
    }

    finish();
  }
}

File: testapp/src/main/java/io/github/juanimoli/strictmode/notifier/ui/ClassInstanceLimitActivity.kt
package io.github.juanimoli.strictmode.notifier.ui

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity

class ClassInstanceLimitActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Post a message and delay its execution for 10 minutes.
        Handler(Looper.getMainLooper()).postDelayed({}, (1000 * 60 * 10).toLong())
        var count = intent.getIntExtra(EXTRA_COUNT, 0)

        if (count > 0) {
            start(this, --count)
        }

        finish()
    }

    companion object {
        private const val EXTRA_COUNT = "EXTRA_COUNT"
        fun start(context: Context, count: Int) {
            val starter = Intent(context, ClassInstanceLimitActivity::class.java)
            starter.putExtra(EXTRA_COUNT, count)
            context.startActivity(starter)
        }
    }
}
